gbpservice/nfp/service_plugins/vpn/drivers/nfp_vpnaas_driver.py

```diff
@@ -0,0 +1,229 @@
+        self.conn.create_consumer(
+            topics.VPN_PLUGIN_TOPIC, self.endpoints, fanout=False)
+        self.conn.consume_in_threads()
+        self.agent_rpc = OCIpsecVpnAgentApi(
+            topics.VPN_AGENT_TOPIC, BASE_VPN_VERSION, self)
+
+    def _get_service_vendor(self, context, vpnservice_id):
+        vpnservice = self.service_plugin.get_vpnservice(
+                context, vpnservice_id)
+        desc = vpnservice['description']
+        # if the call is through GBP workflow,
+        # fetch the service profile from description
+        # else, use 'VYOS' as the service profile
+        if 'service_vendor=' in desc:
+            tokens = desc.split(';')
+            service_vendor = tokens[5].split('=')[1]
+        else:
+            service_vendor = 'VYOS'
+        return service_vendor
+
+    def create_ipsec_site_connection(self, context, ipsec_site_connection):
+        service_vendor = self._get_service_vendor(context,
+                          ipsec_site_connection['vpnservice_id'])
+        self.agent_rpc.vpnservice_updated(
+            context,
+            ipsec_site_connection['vpnservice_id'],
+            rsrc_type='ipsec_site_connection',
+            svc_type=const.SERVICE_TYPE_IPSEC,
+            rsrc_id=ipsec_site_connection['id'],
+            resource=ipsec_site_connection,
+            reason='create', service_vendor=service_vendor)
+
+    def delete_ipsec_site_connection(self, context, ipsec_site_connection):
+        service_vendor = self._get_service_vendor(context,
+                          ipsec_site_connection['vpnservice_id'])
+        self.agent_rpc.vpnservice_updated(
+            context,
+            ipsec_site_connection['vpnservice_id'],
+            rsrc_type='ipsec_site_connection',
+            svc_type=const.SERVICE_TYPE_IPSEC,
+            rsrc_id=ipsec_site_connection['id'],
+            resource=ipsec_site_connection,
+            reason='delete', service_vendor=service_vendor)
+
+    def create_vpnservice(self, context, vpnservice):
+        service_vendor = self._get_service_vendor(context,
+                                                  vpnservice['id'])
+        self.agent_rpc.vpnservice_updated(
+            context,
+            vpnservice['id'],
+            rsrc_type='vpn_service',
+            svc_type=const.SERVICE_TYPE_IPSEC,
+            rsrc_id=vpnservice['id'],
+            resource=vpnservice,
+            reason='create', service_vendor=service_vendor)
+
+    def delete_vpnservice(self, context, vpnservice):
+        service_vendor = self._get_service_vendor(context,
+                                                  vpnservice['id'])
+        self.agent_rpc.vpnservice_updated(
+            context,
+            vpnservice['id'],
+            rsrc_type='vpn_service',
+            svc_type=const.SERVICE_TYPE_IPSEC,
+            rsrc_id=vpnservice['id'],
+            resource=vpnservice,
+            reason='delete', service_vendor=service_vendor)
```

Comment:
I see const.SERVICE_TYPE_IPSEC is not resolved. Can you check them in all the API.